### PR TITLE
Improve auto model registration

### LINKED ISSUE #315: Cannot register a model if there's a non-managed model referring to it
I have these two models, which have worked for me for many months.

```python
class User(AbstractBaseUser, PermissionsMixin):
    username = models.CharField(max_length=250, blank=True)
    email = models.CharField(max_length=100, unique=True, blank=True, null=True)

class UserOrdersView(models.Model):
    user = models.OneToOneField(
        User,
        on_delete=models.DO_NOTHING,
        related_name='user_orders',
        primary_key=True
    )
    # other fields
    class Meta:
        managed = False
        db_table = "users_userordersview"
```

However, when I register the User class, my testcases fail, with the following stack:
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.11/x64/lib/python3.7/site-packages/django/db/models/fields/related_descriptors.py", line 393, in __get__
    rel_obj = self.related.get_cached_value(instance)
  File "/opt/hostedtoolcache/Python/3.7.11/x64/lib/python3.7/site-packages/django/db/models/fields/mixins.py", line 13, in get_cached_value
    return instance._state.fields_cache[cache_name]
KeyError: 'user_orders'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.11/x64/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedTable: relation "users_userordersview" does not exist
LINE 1: ...", "users_userordersview"."churned_date" FROM "users_use...
```

I tried completely removing the view from my app (including the migrations), migrating to a clean database, and registering the user mode, and it worked fine.

Why does auditlog care about the view? The view is not registered. (ironically, I tried registering it and the tests passed).

### LINKED ISSUE #548: pass `include_auto_created=True` to registry.get_models()
This allows including m2m through models.
It could be also a setting value `AUDITLOG_INCLUDE_AUTO_CREATED_MODELS` with default=False to comply with the current implementation.

### PR DESCRIPTION
**Pull request by aqeelat**
closes #315
closes #548 

feat: collect all models including auto created ones and excluding non-managed ones
feat: automatically register m2m fields for models when opting to auto register models

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 - feat: Added pre-log and post-log signals. ([#483](https://github.com/jazzband/django-auditlog/pull/483))
 - feat: Make timestamp in LogEntry overwritable. ([#476](https://github.com/jazzband/django-auditlog/pull/476))
 - feat: Support excluding field names globally when ```AUDITLOG_INCLUDE_ALL_MODELS``` is enabled. ([#498](https://github.com/jazzband/django-auditlog/pull/498))
+- feat: Improved auto model registration to include auto-created models and exclude non-managed models, and automatically register m2m fields for models. ([#550](https://github.com/jazzband/django-auditlog/pull/550))
```
Comment (hramezani) [inline CHANGELOG.md:17]: Shouldn't it be in `Breaking Changes` section?

**Comment by hramezani** [comment]
@aqeelat Thanks for this PR.

Only a question otherwise, LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (aqeelat) [reply CHANGELOG.md:17]: why is this a breaking change?

Comment (hramezani) [reply CHANGELOG.md:17]: Sorry, I was wrong

**Comment by alxvallejo** [comment]
>feat: collect all models including auto created ones and excluding non-managed ones

Why are we excluding non-managed models? Just because a model isn't managed doesn't mean we don't want to track changes to instances of it.

**Pull request merged**